src/main/java/de/bonndan/nivio/input/Indexer.java

```diff
@@ -87,42 +88,45 @@ private void runResolvers(LandscapeDescription input, Landscape landscape) {
         KPIFactory kpiFactory = new KPIFactory();
         landscape.setKpis(kpiFactory.getConfiguredKPIs(input.getConfig().getKPIs()));
 
-        Map<ItemDescription, List<String>> templatesAndTargets = new HashMap<>();
         // read all input sources
-        new SourceReferencesResolver(formatFactory, logger).resolve(input, templatesAndTargets);
+        new SourceReferencesResolver(formatFactory, logger).resolve(input);
 
-        // apply template values to the items
-        new TemplateResolver().processTemplates(input, templatesAndTargets);
+        // apply template values to items
+        new TemplateResolver(logger).resolve(input);
 
-        // read special labels on items and assign the values to fields
-        new LabelToFieldProcessor(logger).process(input, landscape);
+        // resolve links on components to gather more data.
+        new LinksResolver(logger, linkHandlerFactory).process(input);
 
         // mask any label containing secrets
-        new SecureLabelsProcessor().process(input);
+        new SecureLabelsResolver(logger).resolve(input);
 
         // create relation targets on the fly if the landscape is configured "greedy"
-        new InstantItemResolver(logger).processTargets(input);
+        new InstantItemResolver(logger).resolve(input);
+
+        // read special labels on items and assign the values to fields
+        new LabelToFieldResolver(logger).resolve(input);
 
         // find items for relation endpoints (which can be queries, identifiers...)
-        new RelationEndpointResolver(logger).processRelations(input);
+        // KEEP here (must run late after other resolvers)
+        new RelationEndpointResolver(logger).resolve(input);
 
         // add any missing groups
-        new GroupResolver(logger).process(input, landscape);
+        new GroupProcessor(logger).process(input, landscape);
```

Comment:
Since this is all about resolvers and processors. What is the difference between both of them? They both operate on the provided `LandscapeDescription input`. Is the result a modified input which is later applied to the `Landscape`?

Reply:
Maybe the naming can be discussed. Resolvers just modify and extended the input, but not not affect the final output.

Reply:
Why is an input modified if it does not have an effect on the final output?

I am just trying to understand the underlying mechanism ;)

Reply:
First, the data is aggregated in the ComponentDescription objects. This is done sequentially by the different Resolvers. Then the processors do their work and start mapping the input to the final Landscape.

Reply:
Okay thanks, now I understand!